packages/swingset-liveslots/tools/fakeVirtualSupport.js

```diff
@@ -39,6 +44,92 @@ class FakeWeakRef {
   }
 }
 
+/** @typedef {Omit<KVStore,'set' | 'delete'> & Map<string, string>} EnhancedKVStore */
+
+/**
+ * Create a Map backed by a sorted KVStore while keeping the getNextKey method
+ * specific to a KVStore making it mostly compatible with both.
+ *
+ * Iterating over the map while mutating it is "unsupported" (entries inserted
+ * that sort before the current iteration point will be skipped).
+ *
+ * The `size` property is not supported.
+ *
+ * @param {KVStore} fakeStore
+ */
+export function makeEnhancedKVStore(fakeStore) {
+  /** @type {EnhancedKVStore} */
+  const map = harden({
+    ...fakeStore,
+    set(key, value) {
+      fakeStore.set(key, value);
+      return map;
+    },
+    delete(key) {
+      const had = fakeStore.has(key);
+      fakeStore.delete(key);
+      return had;
+    },
+    clear() {
+      for (const key of map.keys()) {
+        fakeStore.delete(key);
+      }
+    },
+    /** @returns {number} */
+    get size() {
+      throw new Error('size not implemented.');
+    },
+    *entries() {
+      for (const key of map.keys()) {
+        yield [key, /** @type {string} */ (fakeStore.get(key))];
+      }
+    },
+    *keys() {
+      /** @type {string | undefined} */
+      let key = '';
+      // eslint-disable-next-line no-cond-assign
+      while ((key = fakeStore.getNextKey(key))) {
+        yield key;
+      }
+    },
```

Comment:
Do we explicitly disavow support for empty string as a key?
```suggestion
    *keys() {
      /** @type {string | undefined} */
      let key = '';
      if (fakeStore.has(key)) {
        yield key;
      }
      // eslint-disable-next-line no-cond-assign
      while ((key = fakeStore.getNextKey(key))) {
        yield key;
      }
    },
```

Reply:
Yeah this surprised me too, but other places seem to rely on this, effectively all the prefix logic.